### PR TITLE
Buidler > Hardhat

### PR DESCRIPTION
**Pull request by Anyhowclick**
Found some failing tests that were not caught before, thanks to more strict type validation checks:
- Bad buy array length
- Negative value for the last LCR test
- Overflow checks done through a SC instead

On the bright side, we now can have just 1 hardhat config file, and specifically overwrite the version and solc runs for specific contracts now!

### CODE REVIEW

File: test/reserveSetup.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ module.exports.setupConversionRateV1 = async function(tokens, admin, operator, a
     await convRatesInst.setBaseRate(tokenAdd, baseBuyRate, baseSellRate, buys, sells, currentBlock, indices, {from: operator});
 
     //set compact data
-    compactBuyArr = [0, 0, 0, 0, 0, 06, 07, 08, 09, 1, 0, 11, 12, 13, 14];
+    compactBuyArr = [0, 0, 0, 0, 0, 6, 7, 8, 9, 10, 11, 12, 13, 14];
```
Comment (ilanDoron) [inline test/reserveSetup.js:76]: one value removed here?

File: coverage.sh [hunk 2]
```diff
@@ -5,21 +5,16 @@ while getopts "v:" arg; do
...
-  yarn buidler coverage --config ./buidlerConfigSol5.js --testfiles "test/sol5/*.js" --solcoverjs ".solcover.js"
+  yarn hardhat coverage --testfiles "test/sol5/*.js" --solcoverjs ".solcover.js"
 else
   echo "Running sol6 coverage..."
-  yarn buidler compile --config buidlerConfigSol5.js &&
-  yarn buidler compile --config buidlerConfigSol4.js
-  cp -R ./artifacts/ ./.tempCoverageArtifacts
-  yarn buidler coverage --config ./buidlerConfigSol6.js --testfiles "test/sol6/*.js" --solcoverjs ".solcover.js"
+  yarn hardhat coverage --testfiles "test/sol6/*.js" --solcoverjs ".solcover.js"
 fi
```
Comment (ilanDoron) [inline coverage.sh:22]: does this create full reports for all sol versions?

File: test/reserveSetup.js (hunk 1 above)
Comment (Anyhowclick) [reply test/reserveSetup.js:76]: Yes, extra value lol. Max array should be 14, but it's 15. Hardhat caught this bug thanks to their stricter input validation checks

File: coverage.sh (hunk 2 above)
Comment (Anyhowclick) [reply coverage.sh:22]: There's some issue with coverage, upgrading to latest version as it provides hardhat support

**Review by ilanDoron — APPROVED** [review]

**Pull request merged**